types/triggerAction.ts

```diff
@@ -0,0 +1,50 @@
+export type TriggerAction =
+| OpenUrl
+| OpenArticleUrl
+| OpenBookmarkedUrl
+| FrequentVisits
+| NewSavedLogin
+| ContentBlocking;
+
+/** List of urls we should match against */
+type UrlParams = Array<string>;
+
+/** List of Match pattern compatible strings to match agaist */
+type UrlPatterns = Array<string>;
+
+interface BaseTrigger {
+    id?: string;
+    params?: UrlParams;
+    patterns?: UrlPatterns;
+}
+
+/** Happens every time the user loads a new URL that matches the provided `hosts` or `patterns` */
+export interface OpenUrl extends BaseTrigger {
+    id: "openURL";
+}
+
+/** Happens every time the user loads a document that is Reader Mode compatible */
+export interface OpenArticleUrl extends BaseTrigger {
+    id: "openArticleURL";
+}
+
+/** Happens every time the user adds a bookmark from the URL bar star icon */
+export interface OpenBookmarkedUrl {
```

Comment:
Why are some of these not extending `BaseTrigger`?

Reply:
They never need urls or patterns.

Reply:
Should they still extend in that case? 🤔